### ISSUE TITLE
Handling redirects

### ISSUE THREAD

**Issue by itaied246** [vietj/vertx-http-proxy#3]
I didn't see any consideration to redirects, how do you handle those?
For example, if my server behind  the proxy response with `302`, you need to modify the response `host` header back to the original.

This is how apache do it:
http://httpd.apache.org/docs/2.0/mod/mod_proxy.html#proxypassreverse

**Comment by vietj** [comment]
@itaied246 sorry I haven't seen your issue, if that is the custom logic of a proxy then it should be implemented otherwise, as the proxy is programmatic it should be implementable.

can you point me where the spec mentions this behavior ?

**Comment by alexlehm** [comment]
I don't think that is mentioned in the http rfc, however it is necessary to run a webserver behind a reverse proxy since otherwise redirects will send the browser to the internal url of the server, which is usually not correct.

**Comment by vietj** [comment]
so that should not be a behavior of the proxy itself but rather a rather a kind of plugin ?

**Comment by alexlehm** [comment]
It has to be configurable somehow, in Apache this is in the mod_proxy module, in Squid I'm not sure.
If we implement a header modification plugin hook, rewrite host could be a standard plugin

**Comment by vietj** [comment]
there are hooks at the moment for allowing header rewrite, look at `ProxyRequestTest#testUpdateRequestHeaders`

**Comment by alexlehm** [comment]
for the Location header we need a header rewrite of the response headers
probably

**Comment by francoisprunier** [comment]
Hi @alexlehm,

That should be doable now if/when my PR #4 is merged.
